connect-inject/health_check_resource.go

```diff
@@ -212,6 +220,11 @@ func (h *HealthCheckResource) registerConsulHealthCheck(client *api.Client, cons
 		},
 	})
 	if err != nil {
+		// Full error looks like:
+		// Unexpected response code: 500 (ServiceID "consulnamespace/svc-id" does not exist)
+		if strings.Contains(err.Error(), fmt.Sprintf("%s\" does not exist", serviceID)) {
```

Comment:
todo: what if the service was deregistered due to agent restart? Look into this.

Reply:
@kschoche so in this case it will miss the event and re-register after reconcile period. I don't think this change makes it worse though because the retries (that happened before this change) happen so fast that they also wouldn't catch the service re-registering.